cEP-0020.md

```diff
@@ -0,0 +1,284 @@
+# Newcomer metrics and Gamification
+
+| Metadata |                                                        |
+| -------- | ------------------------------------------------------ |
+| cEP      | 20                                                     |
+| Version  | 1.0                                                    |
+| Title    | Newcomer metrics and Gamification system               |
+| Authors  | Shrikrishna Singh <mailto:krishnasingh.ss30@gmail.com> |
+| Status   | Proposed                                               |
+| Type     | Process                                                |
+
+## Abstract
+
+This cEP describes how a newcomer at coala will become a developer on the
+basis of their activities in the coala community.
+
+For the implementation of this process, we will incorporate a gamification
+system which will allow us to track newcomer's progress and assign some
+points to each of the activity they do. We will also have different levels
+and badges on the basis of points earned by the newcomers.
+
+So, every newcomer at coala will have to complete the processes involved in
+the gamification system in order to get the **developer badge**.
+
+## Background
+
+### Newcomer Experience
+
+- How to attract more newcomers
+  - How to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala.
+  - Talk to Docs team
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  - How to use feedback to improve coala and processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to the open source community for the first time. Some barriers they face include
+orientation issues that can potentially demotivate newcomers from placing
+their first contribution. On the other hand, gamification is widely used to
+engage and motivate people to accomplish tasks and improve their performance.
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Process involved in the Gamification system
+
+As the goal of this project is to make use of metrics and gamification to
+motivate and support newcomers to overcome the orientation barriers and
+contribute to the community, to achieve this goal we have selected three
+game design elements to help newcomers: Levels, Points, and Badges. For
```

Comment:
I think we need to "Activities" here too.  As I understand it, they are the raw actions.

And I am not understanding the way they are all related.

A worked example would be helpful.

What is the relationship between Activities and Points.  Is each activity given a configurable number of points?